aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -694,6 +699,10 @@ static Tensor & masked_select_out_impl_cpu(Tensor & result, const Tensor & self,
   TORCH_CHECK(self.scalar_type() == result.scalar_type(),
               "masked_select(): self and result must have the same scalar type");
 
+  at::assert_no_internal_overlap(result);
+  at::assert_no_partial_overlap(result, self);
+  at::assert_no_partial_overlap(result, mask);
+
```

Comment:
This file also adds memory overlap support for:
- index_select out= variant (we should add a test for this)
- masked_fill_ (we should add a test for this)
- masked_select out=variant (we should add a test for this)

Reply:
@peterbell10 did you want to add tests for these? I'm fine without them but it is nice to have them to check parity between CPU/CUDA (and to detect regressions)

Reply:
They have tests in #43423 and I moved the asserts to that PR as well.

Reply:
Got it, I didn't realize. Thanks for clarifying